addons/portal/static/src/js/portal_signature.js

```diff
@@ -100,30 +128,39 @@ odoo.define('portal.signature_form', function (require){
                 }
             });
         },
-
-        //--------------------------------------------------------------------------
-        // Private
-        //--------------------------------------------------------------------------
-
-        /**
-         * @private
-         * @returns {Deferred}
-         */
-        _loadTemplates: function () {
-            return ajax.loadXML('/portal/static/src/xml/portal_signature.xml', qweb);
-        },
     });
 
     base.ready().then(function () {
         $('.o_portal_signature_form').each(function () {
             var $elem = $(this);
-            var form = new SignatureForm(null, $elem.data());
+
+            var callUrl = $elem.data('call-url');
+            var nameAndSignatureOptions = {
+                defaultName: $elem.data('default-name'),
+                mode: $elem.data('mode'),
+                signatureRatio: $elem.data('signature-ratio'),
+                signatureType: $elem.data('signature-type'),
+            };
+            var sendLabel = $elem.data('send-label');
+
+            var form = new SignatureForm(null, {
```

Comment:
While waiting for the portal refactoring I am doing (I will have a hard time to rebase), you can use require('root.widget') as parent for this form.

Reply:
ok